lib/proof/domain/entities/download_info_entity.dart

```diff
@@ -2,10 +2,14 @@ class DownloadInfo {
   final bool completed;
   final int contentLength;
   final int downloaded;
+  final bool errorOccurred;
+  final String errorMessage;
 
   DownloadInfo({
```

Comment:
This could be a Sealed class DownloadInfo with 2 childs DownloadDone and DownloadError, it's common to use it in Kotlin.